tests/test_synced_collection.py

```diff
@@ -520,22 +546,30 @@ def test_update_recursive(self, synced_list):
         self.store(data1)
         assert synced_list == data1
 
-        # inavlid data in file
+        # invalid data in file
         data2 = {'a': 1}
         self.store(data2)
         with pytest.raises(ValueError):
             synced_list.load()
 
+        # reset the file
+        self.store([])
+
+    @given(JSONDataStrategy)
+    @example(None)
     def test_reopen(self, synced_list, testdata):
-        synced_list2 = deepcopy(synced_list)
-        synced_list.append(testdata)
-        synced_list.sync()
-        del synced_list  # possibly unsafe
-        synced_list2.load()
-        assert len(synced_list2) == 1
+        synced_list.clear()
+        synced_list2 = deepcopy(synced_list)  # possibly unsafe
+        synced_list2.append(testdata)
         assert synced_list2[0] == testdata
+        del synced_list2  # possibly unsafe
```

Comment:
This deletes the copy instead of the original. *This changes the behavior of the test!* :warning: Too many variables were switched in this test -- I'm not sure if the refactoring you've done here is correct. Please look carefully.